netsim/modules/vlan.py

```diff
@@ -485,8 +480,8 @@ def create_vlan_links(link: Box, v_attr: Box, topology: Box) -> None:
           if 'mode' in intf.vlan and not get_from_box(intf_data,'vlan.mode'):
             intf_data.vlan.mode = intf.vlan.mode            # vlan.mode is inherited from trunk dictionary or parent interface
 
-          if interface_vlan_mode(intf_data,intf_node,topology) == 'bridge':     # Is this VLAN interface in bridge mode?
```

Comment:
This also fixes a bug where vlan.mode was only being checked for the access vlan, instead of the vlan being processed